// Copyright 2018-2020 Drexel University
// Author: Geoffrey Mainland <mainland@drexel.edu>

syntax = "proto3";
package dragonradio.remote;

// See:
//   https://sc2colosseum.freshdesk.com/support/solutions/articles/22000220460-radio-command-and-control-c2-api

enum State {
    OFF = 0;
    BOOTING = 1;
    READY = 2;
    ACTIVE = 3;
    STOPPING = 4;
    FINISHED = 5;
    ERROR = 6;
}

enum RadioCommand {
    START = 0;
    STOP = 1;
    STATUS = 2;
}

message Status {
    State state = 1;
    string info = 2;
}

message MandatedOutcomes {
    string goals = 1;
}

message Environment {
    string environment = 1;
}

message Request {
    double timestamp = 1;
    oneof payload {
        RadioCommand radio_command = 2;
        MandatedOutcomes update_mandated_outcomes = 3;
        Environment update_environment = 4;
    }
}

message Response {
    oneof payload {
        Status status = 1;
    }
}
